moo.js

```diff
@@ -73,7 +78,8 @@
     } else if (Array.isArray(obj)) {
       // sort to help ensure longest match
       var options = sortPatterns(obj)
-      return '(' + options.map(regexpOrLiteral).join('|') + ')'
+      options = options.map(regexpOrLiteral).filter(function(o) { return o !== ERROR })
```

Comment:
It might be cleaner return `null` or `undefined` from `regexpOrLiteral` and filter based on truthiness (`filter(function(o) {return o})`). I'm not sure